src/components/HelloWorld.vue

```diff
@@ -1,175 +1,141 @@
+  axios
+    .put(apiURL + `todos/${id}`, {
+      isComplete: !checkStatus,
     })
-  }
-
-}
-function postCheck(event: Event, id:string, checkStatus:boolean) {
-  axios.put(`https://calm-plum-jaguar-tutu.cyclic.app/todos/${id}`, {
-    'isComplete': !checkStatus
-  }).then((result) => {
-    console.log(result)
-  }).catch((error) => {
-    alert('check was not posted')
-    event.target.checked = !event.target.checked
-    console.log('error check not updated',error)
-  })
-}
-function deleteHandler(id:string) {
-  axios.delete(`https://calm-plum-jaguar-tutu.cyclic.app/todos/${id}`).then((result) => {
-    posts.value = posts.value.filter((item) => {
-      return item._id !== id
+    .then((result) => {
+      console.log(result);
     })
-    console.log(result)
-  }).catch((error) => {
-    console.log('error',error)
-  })
+    .catch((error) => {
+      alert("check was not posted");
+      event.target.checked = !event.target.checked;
+      console.log("error check not updated", error);
+    });
 }
 
-function sort(mode:SortType) {
-  sortMode.value = mode
+function deleteHandler(id: string) {
+  axios
+    .delete(apiURL + `todos/${id}`)
+    .then((result) => {
+      posts.value = posts.value.filter((item) => {
+        return item._id !== id;
+      });
+      console.log(result);
+    })
+    .catch((error) => {
+      console.log("error", error);
+    });
 }
 
+function detailHandler(_id: string) {
+  router.push({ name: "post", params: { id: _id } });
+}
 
-const filteredPosts = computed(() => {
-  if (sortMode.value === 'none') {
-    return posts.value.filter((post) => {
-      return post.todoName.includes(filterQuery.value.toLowerCase())
-  })} else if (sortMode.value === 'ascend') {
-    return posts.value.filter((post) => {
-      return post.todoName.includes(filterQuery.value.toLowerCase())
-  }).sort((a,b) => {
-      if (a.todoName.toLowerCase()  < b.todoName.toLowerCase()) {
-        return -1;
-      }
-      if (a.todoName.toLowerCase() > b.todoName.toLowerCase()) {
-        return 1;
-      }
-      return 0;
-    })
+const sortedPosts = computed(() => {
+  if (sortMode.value === "none") {
+    return filteredPosts.value
+  } else if (sortMode.value === 'ascend') {
+    return filteredPosts.value.slice(0).sort((a,b) => a.todoName.localeCompare(b.todoName)).reverse()
   } else {
-    return posts.value.filter((post) => {
-      return post.todoName.includes(filterQuery.value.toLowerCase())
-    }).sort((a,b) => {
-      if (a.todoName.toLowerCase() < b.todoName.toLowerCase() ) {
-        return 1;
-      }
-      if (a.todoName.toLowerCase()  > b.todoName.toLowerCase() ) {
-        return -1;
-      }
-      return 0;
-    })
-  }
+    return filteredPosts.value.slice(0).sort((a,b) => a.todoName.localeCompare(b.todoName))
   }
+})
 
-)
+const filteredPosts = computed(() => {
+  return posts.value.filter((post) => post.todoName.includes(filterQuery.value.toLowerCase()))
+});
```

Comment:
Huge improvement over the previous computed where everything was handled in one place, but I think you could replace `.slice(0).sort()` with `.toSorted()` since it's now widely supported

Reply:
Thanks, that works but I get a type error for some reason
![image](https://github.com/nicknguyen-ccm/crudapp/assets/156463196/38f3cb24-fc6c-4683-ae0e-851b2c897b03)

Reply:
Probably something in your tsconfig or vite config maybe is set for an older version of Javascript